packages/edit-site/src/store/actions.js

```diff
@@ -298,3 +302,121 @@ export function setIsListViewOpened( isOpen ) {
 		isOpen,
 	};
 }
+
+/**
+ * Reverts a template to its original theme-provided file.
+ *
+ * @param {Object} template The template to revert.
+ */
+export function* revertTemplate( template ) {
+	if ( ! isTemplateRevertable( template ) ) {
+		yield controls.dispatch(
+			noticesStore,
+			'createErrorNotice',
+			__( 'This template is not revertable.' ),
+			{ type: 'snackbar' }
+		);
+		return;
+	}
+
+	try {
+		const fileTemplatePath = addQueryArgs(
+			`/wp/v2/templates/${ template.id }`,
```

Comment:
This is my fault because of how I worded my `apiFetch` suggestion, but I think we should avoid hardcoding the entity's URL.

Looking back at how it's handled in [`getEntityRecord`](https://github.com/WordPress/gutenberg/blob/158d03ce1e98a75fecbe1bd009226cc72e73664b/packages/core-data/src/resolvers.js#L80-L144), we should instead retrieve the entity `baseURL` from state.
I'm not quite sure why the resolver uses `getKindEntities`, but we should be able to do it with the simpler `getEntity`, to end up with something like this (untested code):
```js
const templateEntity = yield controls.select( coreStore, 'getEntity', 'wp_template' );
if ( ! entity ) {
  // dispatch error notice
  return;
}

const fileTemplatePath = addQueryArgs(
  entity.baseURL + / + template.id,
  { context: 'edit', source: 'theme' }
);
// fetch file template, etc.
```